src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -225,13 +226,11 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 
     fn codegen_return_terminator(&mut self, mut bx: Bx) {
         if self.fn_ty.c_variadic {
-            match self.va_list_ref {
-                Some(va_list) => {
+            match self.locals[mir::Local::new(1 + self.fn_ty.args.len())] {
```

Comment:
Could you add a comment. This is clever and I see what you're doing, but it would be good to have some documentation.